python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,395 @@
+    """
+
+    class ModuleWrapper:
+        """The class use use to represent Module and storage module index and
+        Interface information.
+        """
+
+        class Interface:
+            """The class that use to storage module connection information.
+               There are 2 types Interface Input:1 Output:2
+            Parameters
+            ----------
+
+            owner : ModuleWrapper
+                The class that own this interface, in such class there are
+                Module information like index, module name
+
+            itype : integer
+                Interface type, 1 is input interface, 2 is output interface
+
+            name : str/integer
+                Interface name, for input that is string for example "data0"
+                for output that is integer for example 0.
+            """
+
+            def __init__(self, owner, itype, name):
+                self.owner_ = owner
+                self.itype_ = itype
+                self.name_ = str(name)
+                self.dependent_ = []
+
+            def get_name(self):
+                mname = ""
+                if self.owner_:
+                    mname = self.owner_.name_
+
+                return mname, self.name_
+
+            def get_owner_indx(self):
+                return self.owner_.indx_
+
+            def get_dependent_str(self):
+                name = ""
+                for dependent in self.dependent_:
+                    mname, dname = dependent.get_name()
+                    name = name + (mname + ":output(" + dname if self.itype_ == 2 else "")
+                    name = name + (")" if self.itype_ == 2 else mname + ":" + dname)
+                return name
+
+            def add_dependent(self, dependent):
+                """
+                # check if the dependency setting correct.
+                # correct connection are following
+                # 1. global input to module input
+                # 2. module output to next module input
+                # 3. module output to global output
+                """
+                owner_indx = self.get_owner_indx()
+                dep_owner_indx = dependent.get_owner_indx()
+                assert owner_indx != dep_owner_indx, f"can not set self as dependent."
+                assert not (
+                    owner_indx > dep_owner_indx
+                    and not (dependent.itype_ == 2 and dep_owner_indx == 0)
+                ), f"dependent only can be next module interface or global output."
+                assert not (
+                    owner_indx == 0 and dependent.itype_ != 1
+                ), f"global input only can set dependent with module input."
+
+                self.dependent_.append(dependent)
+
+        def __init__(self, indx=0):
+            self.indx_ = indx
+            self.name_ = "mod" + str(indx) if indx else ""
+            self.interfaces_ = {1: {}, 2: {}}
+            self.target_host_ = None
+            self.mod_name_ = "default"
```

Comment:
Can we just use `self.name`?

Reply:
self.mod_name_ get use by graph_executor create and in most case it is "default", so we can not use it here as the name for identification, i would change it into "graph_executor_mod_name" to avoid confuse.

Reply:
IIUC, we can use arbitrary name for graph executor and it's not necessary to be "default". Could you check that? If that's not the case then we probably need to discuss with other folks cuz this doesn't make any sense to me.

Reply:
my fault, this variable used by relay.build not graph_executor,  yes it can be arbitrary name, I would use self.name.